tests/proxy/flake_id_generator/flake_id_generator_test.go

```diff
@@ -0,0 +1,103 @@
+	. "github.com/hazelcast/hazelcast-go-client/config"
+	"github.com/hazelcast/hazelcast-go-client/core"
+	. "github.com/hazelcast/hazelcast-go-client/rc"
+	. "github.com/hazelcast/hazelcast-go-client/tests"
+	"log"
+	"sync"
+	"testing"
+	"time"
+)
+
+var flakeIdGenerator core.FlakeIdGenerator
+var client hazelcast.IHazelcastInstance
+
+const (
+	flakeIdStep             = 1 << 16
+	shortTermBatchSize      = 3
+	shortTermValidityMillis = 3000
+	numRoutines             = 4
+	idsInRoutine            = 100000
+)
+
+func TestMain(m *testing.M) {
+	remoteController, err := NewRemoteControllerClient("localhost:9701")
+	if remoteController == nil || err != nil {
+		log.Fatal("create remote controller failed:", err)
+	}
+	cluster, err := remoteController.CreateCluster("3.9", DefaultServerConfig)
+	remoteController.StartMember(cluster.ID)
+	m.Run()
+	remoteController.ShutdownCluster(cluster.ID)
+}
+
+func TestFlakeIdGeneratorProxy_ConfigTest(t *testing.T) {
+	var myBatchSize int32 = shortTermBatchSize
+	config := hazelcast.NewHazelcastConfig().
+		AddFlakeIdGeneratorConfig(NewFlakeIdGeneratorConfig("gen").SetPrefetchCount(myBatchSize).
+			SetPrefetchValidityMillis(shortTermValidityMillis))
+	client, _ = hazelcast.NewHazelcastClientWithConfig(config)
+	defer client.Shutdown()
+	flakeIdGenerator, _ = client.GetFlakeIdGenerator("gen")
+	// this should take a batch of 3 IDs from the member and store it in the auto-batcher
+	id1, err := flakeIdGenerator.NewId()
+	AssertErrorNil(t, err)
+	// this should take second ID from auto-created batch. It should be exactly next to id1
+	id2, err := flakeIdGenerator.NewId()
+	AssertEqualf(t, err, id1+flakeIdStep, id2, "FlakeIdGenerator NewId() failed")
+
+	time.Sleep(shortTermValidityMillis * time.Millisecond)
+	// this ID should be from a new batch, because the validity elapsed
+	id3, err := flakeIdGenerator.NewId()
+	AssertLessThanf(t, err, id1+flakeIdStep*shortTermBatchSize, id3, "FlakeIdGenerator NewId() failed")
+
+}
+
+func TestFlakeIdGeneratorProxy_ConcurrentlyGeneratedIds(t *testing.T) {
+	wg := sync.WaitGroup{}
+	mu := sync.Mutex{}
+	client, _ := hazelcast.NewHazelcastClient()
+	defer client.Shutdown()
+	flakeIdGenerator, _ = client.GetFlakeIdGenerator("gen")
+	localIdsSlice := make([]map[int64]struct{}, 0)
+	ids := make(map[int64]struct{}, 0)
+	wg.Add(numRoutines)
+	for i := 0; i < numRoutines; i++ {
+		go func() {
+			localIds := make(map[int64]struct{}, 0)
+			for k := 0; k < idsInRoutine; k++ {
+				curId, _ := flakeIdGenerator.NewId()
+				localIds[curId] = struct{}{}
+			}
+			mu.Lock()
+			localIdsSlice = append(localIdsSlice, localIds)
+			mu.Unlock()
+			wg.Done()
+		}()
+	}
+	wg.Wait()
+	for _, localIds := range localIdsSlice {
+		for id, _ := range localIds {
+			ids[id] = struct{}{}
+		}
+	}
+	// if there were duplicate IDs generated, there will be less items in the map than expected
+	AssertEqualf(t, nil, len(ids), numRoutines*idsInRoutine, "FlakeIdGenerator NewId() returned duplicate ids")
+}
```

Comment:
Add a test to ensure that the generator returns the appropriate error when there are no members in the cluster that has a join id smaller than 2^16